### PR TITLE
schedule: fix panic caused by removing tombstone

### LINKED ISSUE #1979: replica checker cause panic
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
after tombstone a store, pd panic:
```
[2019/11/26 17:49:16.196 +08:00] [FATAL] [log.go:294] [panic] [recover="\"invalid memory address or nil pointer dereference\""] [stack="github.com/pingcap/log.Fatal
	/home/jenkins/agent/workspace/build_pd_master/go/pkg/mod/github.com/pingcap/log@v0.0.0-20190715063458-479153f07ebd/global.go:59
github.com/pingcap/pd/pkg/logutil.LogPanic
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/pkg/logutil/log.go:294
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:522
runtime.panicmem
	/usr/local/go/src/runtime/panic.go:82
runtime.sigpanic
	/usr/local/go/src/runtime/signal_unix.go:390
github.com/juju/ratelimit.(*Bucket).Available
	/home/jenkins/agent/workspace/build_pd_master/go/pkg/mod/github.com/juju/ratelimit@v1.0.1/ratelimit.go:251
github.com/pingcap/pd/server/schedule.(*OperatorController).getOrCreateStoreLimit.func1
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/operator_controller.go:738
github.com/pingcap/pd/server/core.(*StoreInfo).IsAvailable
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/core/store.go:95
github.com/pingcap/pd/server/schedule/filter.(*storeLimitFilter).Target
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/filter/filters.go:140
github.com/pingcap/pd/server/schedule/filter.Target
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/filter/filters.go:79
github.com/pingcap/pd/server/schedule/selector.(*ReplicaSelector).SelectTarget
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/selector/selector.go:127
github.com/pingcap/pd/server/schedule/checker.(*ReplicaChecker).selectBestStoreToAddReplica
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/checker/replica_checker.go:145
github.com/pingcap/pd/server/schedule/checker.(*ReplicaChecker).SelectBestReplacementStore
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/checker/replica_checker.go:117
github.com/pingcap/pd/server/schedule/checker.(*ReplicaChecker).checkBestReplacement
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/checker/replica_checker.go:229
github.com/pingcap/pd/server/schedule/checker.(*ReplicaChecker).Check
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/checker/replica_checker.go:110
github.com/pingcap/pd/server/schedule.(*CheckerController).CheckRegion
	/home/jenkins/agent/workspace/build_pd_master/go/src/g  1 e/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/schedule/selector/selector.go:127
github.com/pingcap/pd/server/schedule/checker.(*Repithub.com/pingcap/pd/server/schedule/checker_controller.go:57
github.com/pingcap/pd/server.(*coordinator).patrolRegions
	/home/jenkins/agent/workspace/build_pd_master/go/src/github.com/pingcap/pd/server/coordinator.go:115"]

```

2. What version of PD are you using (`pd-server -V`)?
`725afafd5458097c1ccb9913dbed1529fad0d2fd `

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Fixes #1979.

### What is changed and how it works?
This PR fixes the panic problem by resetting the attached function when removing the store limit.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: server/schedule/operator_controller.go [hunk 1]
```diff
@@ -899,5 +899,6 @@ func (oc *OperatorController) GetLeaderScheduleStrategy() core.ScheduleStrategy
 func (oc *OperatorController) RemoveStoreLimit(storeID uint64) {
 	oc.Lock()
 	defer oc.Unlock()
+	oc.cluster.AttachAvailableFunc(storeID, nil)
```
Comment (lhy1024) [inline server/schedule/operator_controller.go]: why is `nil` rather than `func(){ return false }`

Comment (rleungx) [reply server/schedule/operator_controller.go]: I just reset to the original value when the store limit does not exist.

**Pull request merged**